### PR TITLE
Separate workflow steps

### PR DESCRIPTION
**Pull request by darrell-d**
* Correctly log output via agent using `fmt` vs `log` library
* Build out required directories and config files on the fly
* Separate workflow into distinct steps: Preprocessor, user defined, Postprocessor
* Add python script to build virtual directories for preprocessor
* Some cleanup in variable names

This PR represents a single step along the way of separating concerns with workflows. Future PRs will continue to evolve separating these concerns

### CODE REVIEW

File: pkg/server/workflow.go [hunk 1]
```diff
@@ -45,14 +48,14 @@ func isPath(path string) bool {
 	return false
 }
 
-func (s *server) StartWorkflow(ctx context.Context, request *pb.StartWorkflowRequest) (*pb.WorkflowResponse, error) {
+func (s *server) StartWorkflow(ctx context.Context, request *api.StartWorkflowRequest) (*api.WorkflowResponse, error) {
 
 	id := guuid.New()
-	log.Println("\nStarting workflow")
-	log.Println("ID:" + id.String())
+	fmt.Println("\nStarting workflow")
```
Comment (edmore) [inline pkg/server/workflow.go:54]: @darrell-d why the change from `log` to `fmt`?

Comment (darrell-d) [reply pkg/server/workflow.go:54]: @edmore I wanted the info to display in a cleaner readable format to the user.

Using `log` the output came out printing unicode characters [looking like this](https://imgur.com/a/1sQb8yo)

Where as using `fmt` the output [looks a lot cleaner](https://imgur.com/a/iNxH3fc)

I gathered from googling around that `fmt` (format) was better for displaying any kind of data that a user may want to see where spacing mattered, `log` is great for getting items into a log file since it automatically timestamps the log...great for simple lines of text and going into a log file for an automated process.

I fought with log for a bit to tell it to decode unicode chars, but wasn't getting far.

**Review by jwagenaar — APPROVED** [review]
lgtm

**Review by edmore — APPROVED** [review]
